### PR TITLE
Added mapWithState example. 

### PR DESCRIPTION
**Pull request by rocco408**

### CODE REVIEW

File: streaming/src/main/scala/sparkAtScale/StreamingDirectEmails.scala [hunk 1]
```diff
@@ -79,6 +79,21 @@ object StreamingDirectEmails {
...
+      // This will give a DStream made of state (which is the cumulative count of the keys)
+      val mappingFunc = (email_key: String, email_body: Option[String], state: State[Long]) => {
+          var newMessageCount:Option[Long] = None 
+
+          // Check if state exists
+          if (state.exists) {
+              newMessageCount = Some(state.get + 1L)
+          } else {
+              newMessageCount = Some(0L) 
+          }
```
Comment (ghost) [inline streaming/src/main/scala/sparkAtScale/StreamingDirectEmails.scala]: Minor nit but for code like this I always push the conditional further down (allowing it to be a value): `val newMessageCount = Some(if state.exists (state.get + 1L) else 0L)`

**Review by ghost — APPROVED** [review]

File: dev.conf [hunk 2]
```diff
@@ -3,6 +3,7 @@ sparkAtScale {
   kafkaTopic = "emails"
   file = "data/ratings.dat"
   movie_ids_file = "data/movie_ids.csv"
+  numKafkaKeys = "10"
```
Comment (ghost) [inline dev.conf]: Should this be an int?

Comment (rocco408) [reply dev.conf]: Doesn't have to be, but changed it anyways to make it clear I guess.

File: streaming/src/main/scala/sparkAtScale/StreamingDirectEmails.scala (hunk 1 above)
Comment (rocco408) [reply streaming/src/main/scala/sparkAtScale/StreamingDirectEmails.scala]: done

**Pull request merged**